### ISSUE TITLE
Code fix for CS8421: Pass 'item' as an argument

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#35136]
**Version Used**: Visual Studio 2019 version 16.0

🔗 Related to #35135 

**Steps to Reproduce**:

```csharp
int Method(int x) {
  return Convert();

  static int Convert() => x + 1;
}
```

**Expected Behavior**:

* CS8421: A static local function cannot contain a reference to 'x'.
* A code fix is presented which offers to explicitly pass `x` as an argument.

**Actual Behavior**:

The error is reported, but the code fix must be applied manually.

**Comment by genlu** [comment]
Assigned to @GovernessS

**Comment by sharwell** [comment]
@genlu If @GovernessS replies in this issue, I can assign the issue

**Comment by sharwell** [comment]
In design review, the team agreed this is a reasonable fix to provide for this compiler error. It is already assigned to @GovernessS for implementation 👍
